Meet The Teacher/controllers/HeadOfYearController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/*
+Gibbon, Flexible & Open School System
+Copyright (C) 2010, Ross Parker
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+class HeadOfYearController implements PESAPIController
+{
+
+	private $db;
+
+    //Need the distinct in both cases since it could be possible IN has different levels assigned.
+	private $sqlCommands = array(
+        "GetAll" => "
+			select
+				gse.gibbonPersonID as 'StudentID',
+				gyg.gibbonYearGroupID as 'GroupID',
+				gyg.name as 'GroupName',
+				gyg.gibbonPersonIDHOY as 'TeacherID'
+			from gibbonYearGroup gyg
+			inner join gibbonStudentEnrolment gse on gse.gibbonYearGroupID = gyg.gibbonYearGroupID
```

Comment:
Hi Jim, interesting to see gibbonStudentEnrolment in a staff query. Seems like this could be dropped, unless I've missed something...?

Reply:
I could only see that link to gibbonPerson from gibbonYearGroups. I didn't see a column for it. Is there a better table for it?

The reason that's used is to get the list of students inner joined to the member of staff. That way I can pull both the teacher and the link between them and their students at the same time. Saves you having to make some kinda funky group for them.